### PR TITLE
Fix issues for Dereference

### PR DESCRIPTION
**Pull request by joy-yao**

### CODE REVIEW

**Comment by joy-yao** [comment]
Fixes https://github.com/facebook/presto/issues/3823, https://github.com/facebook/presto/issues/3824 and https://github.com/facebook/presto/issues/3825,

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -284,6 +296,57 @@ public void testRowFieldAccessorInJoin()
...
 
+    @Test
+    public void testDereferenceInSubquery()
+            throws Exception
+    {
+        assertQuery("" +
+                        "SELECT x " +
+                        "FROM (" +
+                        "   SELECT a.x" +
+                        "   FROM (VALUES 1,2,3) a(x)" +
```
Comment (cberner) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: space after commas?

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 2]
```diff
@@ -535,6 +598,13 @@ public void testAggregationWithSomeArgumentCasts()
     }
 
     @Test
+    public void testAggregationWithHaving()
+            throws Exception
+    {
+        assertQuery("SELECT a, count(1) FROM (VALUES 1,2,3,2) t(a) GROUP BY a HAVING count(1) > 1", "SELECT 2, 2");
```
Comment (cberner) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: space after commas?

**Comment by cberner** [comment]
Looks good

**Pull request closed without merging**